upload/system/library/Packetery/Vendor/AddEditPage.php

```diff
@@ -0,0 +1,340 @@
+
+class AddEditPage {
+
+	/** @var VendorRepository */
+	private $vendorRepository;
+
+	/** @var CarrierRepository */
+	private $carrierRepository;
+
+	/** @var string */
+	private $country;
+
+	/** @var \Language */
+	private $language;
+
+	/** @var \Session $session */
+	private $session;
+
+	/** @var array */
+	private $vendors = [];
+
+	/** @var array */
+	public $hasErrors = false;
+
+	/**
+	 * @param VendorRepository  $vendorRepository
+	 * @param CarrierRepository $carrierRepository
+	 * @param string            $country
+	 * @param \Language         $language
+	 * @param \Session          $session
+	 */
+	public function __construct(
+		VendorRepository  $vendorRepository,
+		CarrierRepository $carrierRepository,
+		                  $country,
+		\Language         $language,
+		\Session          $session
+	) {
+		$this->vendorRepository = $vendorRepository;
+		$this->carrierRepository = $carrierRepository;
+		$this->country = $country;
+		$this->language = $language;
+		$this->session = $session;
+		$this->hasErrors = false;
+		$this->init();
+	}
+
+	/**
+	 * @return void
+	 */
+	public function init() {
+		$vendors = $this->vendorRepository->getVendorsByCountry($this->country);
+		$carriers = $this->carrierRepository->getCarriersByCountry($this->country);
+
+		//filter out already used in vendors
+		$usedCarrierIds = [];
+		if (!empty($vendors)) {
+			$usedCarrierIds = array_map(
+				static function ($vendor) {
+					return $vendor['carrier_id'];
+				},
+				$vendors);
+		}
+		$vendors = array_filter(
+			$carriers,
+			static function ($carrier) use ($usedCarrierIds) {
+				return !in_array($carrier['id'], $usedCarrierIds, true);
+			}
+		);
+
+		$this->vendors = $vendors;
+	}
+
+	/**
+	 * @param array $weightRules
+	 *
+	 * @return array
+	 */
+	private function removeEmptyWeightRules(array $weightRules) {
+		return array_filter($weightRules, static function ($rule) {
+			return !(empty($rule['max_weight']) && empty($rule['price']));
+		});
+	}
+
+	/**
+	 * @param string $countryCode
+	 *
+	 * @return array
+	 */
+	public function getInternalVendorsByCountry($countryCode) {
+		$vendors = $this->vendorRepository->getInternalVendorsByCountry($countryCode);
+		$usedVendorGroups = $this->vendorRepository->getUsedVendorGroupsByCountry($countryCode);
+
+		$internalVendors = [];
+		foreach ($vendors as $vendor) {
+			if (in_array($vendor['id'], $usedVendorGroups, true) || ($vendor['id'] === 'zpoint' && in_array('', $usedVendorGroups, true))) {
+				continue;
+			}
+			$internalVendors[] = [
+				'vendor_id' => $vendor['id'],
+				'name'      => $this->language->get($vendor['name']),
+			];
+		}
+
+		return $internalVendors;
+	}
+
+	/**
+	 * @param array $formData
+	 *
+	 * @return array
+	 */
+	public function getAddVendorFormValidationErrors(array $formData) {
+		$errors = [];
+		$weightRulesErrors = [];
+
+		if ($formData['action'] === ControllerExtensionShippingZasilkovna::ACTION_ADD_VENDOR) {
+			if (empty($formData['vendor'])) {
```

Comment:
Pokud ve formuláři nevyberu žádného dopravce, tak $formData['vendor'] bude "-1".  Takže to projde touto podmínkou a vytvoří se vendor , kde carrier_id je -1.